### PR TITLE
nsync

### PR DESCRIPTION
**Pull request by njzjz**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

btw, I commit 1.20.1 but not the newest version because it's [dependency of TensorFlow 1.13](https://github.com/tensorflow/tensorflow/blob/r1.13/tensorflow/workspace.bzl).

### CODE REVIEW

File: recipes/nsync/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      run-once initialization
+      waitable counter (useful for barriers)
+      waitable bit (useful for cancellation, or other conditions)
+  doc_url: https://github.com/google/nsync
+  dev_url: https://github.com/google/nsync
+
+extra:
+  recipe-maintainers:
+    - njzjz
+
```
Comment (njzjz) [inline recipes/nsync/meta.yaml]:
```suggestion
```

File: recipes/nsync/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  url: https://github.com/google/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: 692f9b30e219f71a6371b98edd39cef3cbda35ac3abc4cd99ce19db430a5591a
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - cmake >=3.0
```
Comment (njzjz) [inline recipes/nsync/meta.yaml]:
```suggestion
    - cmake >=3.0
    - make
```

File: recipes/nsync/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+source:
+  url: https://github.com/google/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: 692f9b30e219f71a6371b98edd39cef3cbda35ac3abc4cd99ce19db430a5591a
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (njzjz) [inline recipes/nsync/meta.yaml]:
```suggestion
    - {{ compiler('c') }}
    - {{ compiler('cxx') }}
```

**Comment by njzjz** [comment]
@conda-forge/help-c-cpp Please review. Thank you!

**Comment by njzjz** [comment]
@conda-forge/staged-recipes Is there anyone to review?

File: recipes/nsync/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  skip: True  # [win and vc<14]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake >=3.0
+
+test:
+  commands:
```
Comment (chrisburr) [inline recipes/nsync/meta.yaml]: Can you add a test for the Windows build to ensure empty packages aren't uploaded?

File: recipes/nsync/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+build:
+  number: 0
+  skip: True  # [win and vc<14]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake >=3.0
```
Comment (chrisburr) [inline recipes/nsync/meta.yaml]:
```suggestion
    - cmake >=3.0
    - make  # [not win]
```

File: recipes/nsync/meta.yaml (hunk 4 above)
Comment (njzjz) [reply recipes/nsync/meta.yaml]:
```suggestion
  commands:
    - if not exist ${PREFIX}/lib/libnsync.a exit 1  # [win]
    - if not exist ${PREFIX}/lib/libnsync_cpp.a exit 1  # [win]
```

File: recipes/nsync/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    - make  # [not win]
+
+test:
+  commands:
+    - if not exist %LIBRARY_LIB%\\libnsync.a exit 1  # [win]
```
Comment (njzjz) [inline recipes/nsync/meta.yaml]:
```suggestion
    - if not exist %LIBRARY_LIB%\\libnsync.lib exit 1  # [win]
```

File: recipes/nsync/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    - if not exist %LIBRARY_LIB%\\libnsync_cpp.a exit 1  # [win]
```
Comment (njzjz) [inline recipes/nsync/meta.yaml]:
```suggestion
    - if not exist %LIBRARY_LIB%\\libnsync_cpp.lib exit 1  # [win]
```

File: recipes/nsync/meta.yaml (hunk 4 above)
Comment (njzjz) [reply recipes/nsync/meta.yaml]: > Can you add a test for the Windows build to ensure empty packages aren't uploaded?

have fixed them

**Comment by njzjz** [comment]
Have fixed them.
